include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_coupled_solver.cpp

```diff
@@ -176,8 +176,8 @@ TimeIntBDFCoupled<dim, Number>::do_timestep_solve()
     solution_np = solution_last_iter;
   }
 
-  // update viscosity model
-  if(this->param.viscosity_is_variable())
+  // explicit viscosity update
+  if(this->param.viscosity_is_variable() and not this->param.viscous_term_is_nonlinear())
   {
```

Comment:
note that if we go for `TreatmentOfVariableViscosity::Implicit` and hence `this->param.viscous_term_is_nonlinear()` is `true`, the residual update in the Newton solver updates the viscosity prior to solving.